client/lib/domains/email-forwarding/index.js

```diff
@@ -7,4 +13,29 @@ function emailForwardingPlanLimit( plan ) {
 	return ( isBusiness( plan ) ? 100 : 5 );
 }
 
-module.exports = { emailForwardingPlanLimit };
+function validateAllFields( fieldValues ) {
+	return mapValues( fieldValues, ( value, fieldName ) => {
+		const isValid = validateField( {
+			value,
+			name: fieldName
+		} );
+
+		return isValid ? [] : [ 'Invalid' ];
+	} );
+}
+
+function validateField( { name, value } ) {
+	switch ( name ) {
+		case 'mailbox':
+			return value.length > 0 && /^([a-z0-9\._\+\-])*$/i.test( value ) && ! /(^\.)|(\.{2,})|(\.$)/.test( value );
```

Comment:
Actually, looking further, we also have an upper limit on the length: 64 characters (try submitting a very long `mailbox`). So taking that into account, the final final version of this regexp: `/^[a-z0-9._+-]{1,64}$/i`